### PR TITLE
[WIP] Update SDSS module to SDSS-IV URLs

### PR DESCRIPTION
**Pull request by weaverba137**
@bsipocz, @keflavich I'm available to help update the SDSS subpackage.  Here's a start; I'm sure there's plenty more to do.  In particular, `sdss/__init__.py` contains this warning:
```python
warnings.warn("Experimental: SDSS has not yet been refactored to have its API "
              "match the rest of astroquery (but it's nearly there).")
```
What needs to be done to refactor the API?

### CODE REVIEW

**Comment by keflavich** [comment]
Maybe we just need to remove that warning... it looks like the API was already refactored but we forgot to remove the note.

**Comment by weaverba137** [comment]
Hmm, I think there may have been a recent change to the astropy bot, that is no longer finding the change log file.

**Comment by weaverba137** [comment]
I just came down with a cold, but I'll try to get this done this week.

**Comment by weaverba137** [comment]
Working on the rebase now...

**Comment by weaverba137** [comment]
OK, merge when ready.

**Comment by keflavich** [comment]
The failures are real; changing the default database has changed some of the test results.  Could you either update the test data or the tests themselves?  I can assist with that if needed.

File: astroquery/sdss/field_names.py [hunk 1]
```diff
@@ -66,21 +66,21 @@ def _columns_json_to_table(jsonobj):
 # below here are builtin data files
 def _load_builtin_table_fields():
     key1 = ('PhotoObjAll',
-            'http://skyserver.sdss.org/dr12/en/tools/search/x_sql.aspx')
+            conf.skyserver_baseurl + '/dr12/en/tools/search/x_sql.aspx')
```
Comment (bsipocz) [inline astroquery/sdss/field_names.py]: I wonder whether it makes any sense to remove hardwiring DR12 from here. Are the field names same for all the DRs?

File: astroquery/sdss/core.py [hunk 2]
```diff
@@ -102,9 +103,9 @@ def query_crossid_async(self, coordinates, obj_names=None,
         """
 
         if (not isinstance(coordinates, list) and
-                not isinstance(coordinates, Column) and
```
Comment (bsipocz) [inline astroquery/sdss/core.py]: keep this logical indent

File: astroquery/sdss/field_names.py (hunk 1 above)
Comment (weaverba137) [reply astroquery/sdss/field_names.py]: The database column names should be pretty much the same from DR8 onward.  However, the URL paths changed in DR12, mostly `.asp` to `.aspx`.

**Comment by weaverba137** [comment]
@keflavich, could you please point out which failure you are talking about?  The only failure I can see is https://travis-ci.org/astropy/astroquery/jobs/471448405, which uses really, really old versions of everything.

**Comment by keflavich** [comment]
These are the failures I see locally:
/Users/adam/anaconda/envs/astropy36mpl2/lib/python3.6/urllib/request.py:650: HTTPError --> https://bpaste.net/show/2fe0a26d3b6b
/Users/adam/anaconda/envs/astropy36mpl2/lib/python3.6/urllib/request.py:650: HTTPError --> https://bpaste.net/show/99be1a4c6f2a
/Users/adam/anaconda/envs/astropy36mpl2/lib/python3.6/urllib/request.py:650: HTTPError --> https://bpaste.net/show/f8c9cc19d1f8
astroquery/sdss/tests/test_sdss_remote.py:102: AssertionError --> https://bpaste.net/show/4034d6d5c692

I'm not sure why [this](https://travis-ci.org/astropy/astroquery/jobs/471448405) is failing.  The failures I'm quoting are remote failures; you need to run `python setup.py test -P sdss --remote-data` to see them.  We aren't running the remote tests on travis in general (to avoid spamming the servers); we only run them as cron jobs or on merges into master.

File: astroquery/sdss/core.py (hunk 2 above)
Comment (weaverba137) [reply astroquery/sdss/core.py]: I have restored the original indentation.

**Comment by weaverba137** [comment]
OK, at least some (in fact 3/4) of the problems are caused by `IMAGING_URL_SUFFIX` that needed to be updated for post DR12 image paths.  I just pushed a fix for that.  I'll have to look at the other one by hand.

**Comment by weaverba137** [comment]
When comparing tables row by row, as in `row in xid`, how many decimal points of precision are used to compare floating point numbers?  Because as far as I can tell the only difference in the data is that the RA and Dec differ at the order 1e-8 level or smaller.

**Comment by weaverba137** [comment]
PS, the specific test I'm referring to is `test_sdss_specobj` in `test_sdss_remote.py`.

**Comment by keflavich** [comment]
It looks like exact comparison is being used now; it would make sense to switch to `np.testing.assert_almost_equal` or something similar.  Of course, you can't do `row in table` with that sort of setup, so it would require some restructuring.  Are you up to that?  Anything you can do to make the tests saner & easier to understand would be very welcome.

**Comment by weaverba137** [comment]
@keflavich, see the latest push.

**Comment by keflavich** [comment]
Looks great, and passes locally.  I'll merge when Travis is ready.

**Comment by weaverba137** [comment]
The one remaining Travis failure is due to a pytest incompatibility.  Has that been fixed on master?

**Comment by keflavich** [comment]
We're still working on it. Sorry about the delay.

**Comment by bsipocz** [comment]
There are some related looking failures with the old astropy, though I suspect it's only string representation kind of differences. We either need to modify the tests or simply just xfail them when astropy or numpy is older than a certain version.

And a rebase wouldn't hurt given the last merge commit by @keflavich. (note: using github is not the greatest choice to resolve conflicts).

**Comment by keflavich** [comment]
Python 3.4 + astropy 1.3 + numpy1.10 travis-ci testing was deprecated with this PR.  I suspect any of those versions will still work with astroquery, but we're not going to test them any more.

**Comment by weaverba137** [comment]
Thank you for sticking with this!

**Comment by bsipocz** [comment]
> Python 3.4 + astropy 1.3 + numpy1.10 

I'm adding these to the docs and requirements files, too.

**Pull request merged**